mono/metadata/w32file-unix.c

```diff
@@ -4558,13 +4563,18 @@ static _wapi_drive_type _wapi_drive_types[] = {
 };
 
 #if __linux__
-static guint32 _wapi_get_drive_type(long f_type)
+static guint32 _wapi_get_drive_type(const struct statfs *stat1)
 {
-	_wapi_drive_type *current;
+	const _wapi_drive_type *current;
 
 	current = &_wapi_drive_types[0];
 	while (current->drive_type != DRIVE_UNKNOWN) {
-		if (current->fstypeid == f_type)
+		// f_type is the platform-specific internal __statfs_word, which we
+		// prefer not to utter, but still use a temporary with exact matching type.
+		// We don't want to be surprised by varying zero-extension vs. sign-extension.
+		struct statfs stat2;
+		stat2.f_type = current->fstypeid;
```

Comment:
This line is still imperfect. We are assuming the conversion of unsigned long to f_type is lossless.

I have one more idea here, and that is to never store our data.
Instead of the array of constants, we can generate an if/else ladder or switch.
So it'd look like switch (stat1->f_type) case XX.

However, that makes me think we should just __statfs_word as a lesser evil.

I'm going to read the header/manpage/bing again. This shouldn't be so difficult.

Reply:
The man page acknowledges our exact situation:

"conundrum when trying to copy or compare these fields to local variables in a program"

and says "Using unsigned int for such variables suffices on most systems."

"most systems"?